PurchasesCoreSwift/Public/EntitlementInfo.swift

```diff
@@ -0,0 +1,346 @@
+//
+//  EntitlementInfo.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 6/25/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+/**
+ Enum of supported stores
+ */
+@objc(RCStore) public enum Store: Int {
+    /// For entitlements granted via Apple App Store.
+    @objc(RCAppStore) case appStore = 0
+    /// For entitlements granted via Apple Mac App Store.
+    @objc(RCMacAppStore) case macAppStore = 1
+    /// For entitlements granted via Google Play Store.
+    @objc(RCPlayStore) case playStore = 2
+    /// For entitlements granted via Stripe.
+    @objc(RCStripe) case stripe = 3
+    /// For entitlements granted via a promo in RevenueCat.
+    @objc(RCPromotional) case promotional = 4
+    /// For entitlements granted via an unknown store.
+    @objc(RCUnknownStore) case unknownStore = 5
+}
+
+/**
+ Enum of supported period types for an entitlement.
+ */
+@objc(RCPeriodType) public enum PeriodType: Int {
+    /// If the entitlement is not under an introductory or trial period.
+    @objc(RCNormal) case normal = 0
+    /// If the entitlement is under a introductory price period.
+    @objc(RCIntro) case intro = 1
+    /// If the entitlement is under a trial period.
+    @objc(RCTrial) case trial = 2
+}
+
+@objcMembers @objc(RCEntitlementInfo) public class EntitlementInfo: NSObject {
+    /**
+     The entitlement identifier configured in the RevenueCat dashboard
+     */
+    public let identifier: String
+
+    /**
+     True if the user has access to this entitlement
+     */
+    public let isActive: Bool
+
+    /**
+     True if the underlying subscription is set to renew at the end of
+     the billing period (expirationDate). Will always be True if entitlement
+     is for lifetime access.
+     */
+    public let willRenew: Bool
+
+    /**
+     The last period type this entitlement was in
+     Either: RCNormal, RCIntro, RCTrial
+     */
+    public let periodType: PeriodType
+
+    /**
+     The latest purchase or renewal date for the entitlement.
+     */
+    public let latestPurchaseDate: Date? // TODO: This used to be NON-NULL
```

Comment:
could you elaborate on why they're optional now?

Reply:
Our date parsing code returns either a date or a nil

Reply:
ouch

Reply:
ok, this is a tricky case, but we might as well solve it since it's bound to come up again: 
- in obj-c, we're declaring this as non-null, but it's actually nullable. 
- it's nullable in the rest API, so we should try to reflect that (i.e.: the bug is in the declaration, not the implementation)

Currently, this means: 
- for obj-c clients, this is bad, but it's objc, what's nullability anyway in obj-c land. 
- for swift clients, this is **terrible.** The property will show up as non-null, they won't be able to check for nullability, but if it's `nil`, app crashes. 

So I'd say we should just go ahead and make it optional, without even the comment. It's technically a change of the API, but it's more of a _fix_ than an actual change. 
We had a similar thing happen with the `shared` instance of [purchases recently](https://github.com/RevenueCat/purchases-ios/pull/508)

Reply:
I 100% agree with this approach. It changes the Swift API, but it is a fix. I also think we should talk about revving RC a major version, but with enough comms that folks understand it was additive for ObjC, and some small API fixes around nullablility for swift.